src/AddNewRow.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { EventHandler, SyntheticEvent } from "react";
+import { convertTimeStampToDate } from "./util";
+import { Form } from "react-bootstrap";
+import { GoingIcon } from "./resources/GoingIcon";
+import { NotGoingIcon } from "./resources/NotGoingIcon";
+
+type AddNewRowProps = {
+  dates: number[],
+  availableDates: number[],
+  setAvailableDates: React.Dispatch<React.SetStateAction<number[]>>,
+  handleNameUpdate: React.ChangeEventHandler,
+  name: string,
+}
+
+
+
+export const AddNewRow: React.FC<AddNewRowProps> = ({
+  dates,
+  availableDates,
+  setAvailableDates,
+  handleNameUpdate,
+  name,
+}) => {
+
+  const isKeyboardEvent = (event: React.MouseEvent | React.ChangeEvent | React.KeyboardEvent | React.FormEvent): event is React.KeyboardEvent => {
+    return (event as React.KeyboardEvent).key !== undefined;
+  }
+
+  const handleClickEvent: EventHandler<SyntheticEvent> = (e: React.MouseEvent | React.ChangeEvent | React.KeyboardEvent | React.FormEvent) => {
+
+    if (e.type !== 'click' && e.type !== 'keydown' && e.type !== 'change') return;
```

Comment:
I added the `e.type !== 'change'` to this if statement because the change event type was triggering this event so why not use it? This is what I most wanted you to review for me.